src/cpp/library/CUDA/cuda_runtime_api.h

```diff
@@ -279,6 +285,7 @@ cudaDeviceList_t *cudaDeviceList = NULL;
 //Insert a device
 void cudaDeviceInsert(int device)
 {
+  __ESBMC_atomic_begin();
```

Comment:
There is a return statement in the middle of this function. We need to run __ESBMC_atomic_end() before it, too.

Reply:
lt will fail the **Codacy Static Code Analysis**.

Reply:
My understanding is that if we don't do it then the atomic block will be kept open for ESBMC. Is that correct?

Reply:
You are correct, @fbrausse.

@XLiZHI: you should call atomic-end before the return statement.

Reply:
@XLiZHI If adding the atomic-end before return causes Codacy to go wild, that's not a big problem. It's better to make ESBMC faster by avoiding interleavings through atomic blocks than to listen to an inferior static-analysis tool. ;) It's not like Codacy doesn't help, it's just not always right.